src/Analyzers/Core/CodeFixes/UseCoalesceExpression/AbstractUseCoalesceExpressionForIfNullStatementCheckCodeFixProvider.cs

```diff
@@ -46,15 +46,23 @@ protected override Task FixAllAsync(
             var ifStatement = diagnostic.AdditionalLocations[1].FindNode(getInnermostNodeForTie: true, cancellationToken);
             var whenTrueStatement = diagnostic.AdditionalLocations[2].FindNode(getInnermostNodeForTie: true, cancellationToken);
 
+            var left = expressionToCoalesce.WithoutTrivia();
+
+            if (ShouldAddExplicitCast(syntaxFacts, semanticModel, expressionToCoalesce,
+                whenTrueStatement, out var castTo, cancellationToken))
```

Comment:
i'd have this just return the castTo type as the return value.  and this can then just check that for null, and cast if not.

Reply:
Done